common.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+set -e
+
+function wait_weaviate() {
+  local port="${1:-8080}" # Set default port to 8080 if $1 is not provided
+  echo "Wait for Weaviate to be ready"
+  for _ in {1..120}; do
+    if curl -sf -o /dev/null localhost:$port/v1/.well-known/ready; then
+      echo "Weaviate is ready"
+      return 0
+    fi
+
+    echo "Weaviate is not ready on $port, trying again in 1s"
+    sleep 1
+  done
+  echo "ERROR: Weaviate is not ready in port ${port} after 120s"
+  exit 1
+}
+
+function shutdown() { 
+  echo "Cleaning up resources..."  
+  container_count=$(docker container ls -aq | wc -l)
+  if [ "$container_count" -gt 0 ]; then
+    # Place the command you want to execute here
+    echo "There are $container_count containers."
+    docker container rm -f $(docker container ls -aq)
+  fi
+    
+  compose_files=$(ls apps/weaviate/docker-compose-*.yml)
+  for file in $compose_files; do
+    docker compose -f "$file" down --remove-orphans
+  done
+  
+  rm -rf apps/weaviate/data* || true    
+  rm -rf workdir
+}
+trap 'shutdown; exit 1' SIGINT ERR
```

Comment:
Something that would be missing with this implementation of shutdown it's:
```
echo "Showing logs:"
docker compose -f apps/weaviate/docker-compose-replication.yml logs weaviate-node-1 weaviate-node-2 weaviate-node-3
```
which is pretty useful for debugging failed cases, although I have been thinking on enabling logging for all jobs and storing them in Artifacts, maybe now it's the moment.

Reply:
good idea